src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerCreateHearingSummary.java

```diff
@@ -46,21 +48,38 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
                 .name("Create hearing summary")
                 .showSummary()
                 .aboutToStartCallback(this::aboutToStart)
+                .aboutToSubmitCallback(this::aboutToSubmit)
                 .showEventNotes()
                 .grant(CREATE_READ_UPDATE_DELETE, COURT_ADMIN_CIC, SUPER_USER)
                 .grantHistoryOnly(SOLICITOR));
         createHearingSummary.addTo(pageBuilder);
         hearingTypeAndFormat.addTo(pageBuilder);
+        hearingVenues.addTo(pageBuilder);
     }
 
     public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(CaseDetails<CaseData, State> details) {
         var caseData = details.getData();
         DynamicList hearingDateDynamicList = hearingService.getHearingDateDynamicList(details);
         caseData.getHearingSummary().setHearingSummaryList(hearingDateDynamicList);
+        caseData.setCurrentEvent(CASEWORKER_CREATE_HEARING_SUMMARY);
 
         return AboutToStartOrSubmitResponse.<CaseData, State>builder()
             .data(caseData)
             .build();
     }
 
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
+        final CaseDetails<CaseData, State> details,
+        final CaseDetails<CaseData, State> beforeDetails
+    ) {
+        var caseData = details.getData();
+        caseData.setCurrentEvent("");
```

Comment:
Do we need to set currentEvent?

Reply:
I have declared new field currentEvent, which holds have of current event and can be accessed as page level. so I am setting it on abouttostart and clearing it on aboutToSubmit.
I couldn't figured out another way to do it